File: C++ Code/palindrome.c
#include<stdio.h>
 void main()
 {
     int n,reversed_integer=0,original_integer,remainder;
     printf("Enter the value of n=");
     scanf("%d",&n);
     original_integer=n;
     while(n!=0)
     {
         remainder=n%10;
         reversed_integer=reversed_integer*10+remainder;
         n/=10;
     }
     if(original_integer==reversed_integer)
        printf("\n %d is a palindrome.",original_integer);
     else
        printf("\n %d is not a palindrome.",original_integer);
 }

File: JAVA code n material/SelfPratice/src/Problems/Palindrome.java
package Problems;

import java.util.Scanner;

public class Palindrome {
	
public boolean isPalindrome(int x) {
        
        int remainder, reverse = 0, original = x;
        
        if ( x > 0 ) {
        	
        	while ( x != 0 ) {
                
                remainder = x % 10;
                reverse = ( reverse * 10 ) + remainder;
                x /= 10;
                
            }
        	  
        }
        
        if ( reverse == original )
            return true;
        else 
            return false;
        
    }

	public static void main(String[] args) {
		
		Palindrome palindrome = new Palindrome ();
		Scanner sc = new Scanner ( System.in );
		
		System.out.print ( "Enter the number:-" );
		int x = sc.nextInt ();
		
		System.out.print ( palindrome.isPalindrome (x) );

	}

}
